File: wk1/mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // gets height from user, from 1 to 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // loop for each row, with /n at the end of each
    for (int row = 0; row < height; row++)
    {
        // spaces to the left
        for (int space = 0; space < height - 1 - row; space++)
        {
            printf(" ");
        }
        // hashes on the left
        for (int hash = row + 1; hash > 0; hash--)
        {
            printf("#");
        }
        // spaces in the center
        {
            printf("  ");
        }
        // hashes to the right
        for (int hash = row + 1; hash > 0; hash--)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: wk6/sentimental-mario-less/mario.py
from cs50 import get_int

# gets height of pyramid from user, accepting 1-8
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# for each row up to user-specified height
for row in range(height):
    # print spaces before pyramid
    print(" " * (height - 1 - row), end="")

    # print hashes
    print("#" * (row + 1), end="")

    # new line after each row
    print()
